lib/docker-file.ts

```diff
@@ -26,10 +26,45 @@ async function readDockerfileAndAnalyse(
   return analyseDockerfile(contents);
 }
 
+function resolveVariables(
+  text: string,
+  instruction: Instruction,
+  dockerfile: Dockerfile,
+): string {
+  const variables = instruction.getVariables();
+  if (variables) {
+    const resolvedVariables = variables.reduce((resolvedVars, variable) => {
+      const line = variable.getRange().start.line;
+      const name = variable.getName();
+      resolvedVars[name] = dockerfile.resolveVariable(name, line);
+      return resolvedVars;
+    }, {});
+    Object.keys(resolvedVariables).forEach((variable) => {
+      text = text.replace(`\$\{${variable}\}`, resolvedVariables[variable]);
```

Comment:
The library claims to do "variable lookup and resolution" yet here we do a rather crude text.replace to resolve variables ourself - Is there no better way to do this? I looked briefly at the library but the documentation seems to be rather "limited".

Reply:
agree with @mladkau, try to check if we can utilize the library to do this

Reply:
Couldn't find any equivalent, even for variable lookup.
I updated the regexp to support either `$variable_name` or `${variable_name}` forms.
https://docs.docker.com/v17.09/engine/reference/builder/#environment-replacement